MoreLinq/GroupPartition.cs

```diff
@@ -0,0 +1,144 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2009 Atif Aziz. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Partitions a grouping by Boolean keys into a projection of true
+        /// elements and false elements, respectively.
+        /// </summary>
```

Comment:
>>  for the particular boolean overloads, it should be documented which of the resultSelector parameters corresponds to true and false.
>
>This is already done: a projection of true elements and false elements, respectively. Perhaps it's not obvious but “respectively” means in the aforementioned order.

Right, sorry. I meant that this needs to be in the `<param />` entry. 

> On that note, this is different F# where, for example, List.partition returns a couple of false (first) & true (second) parts. Should we align? It does mean though that in the case of bool? keys, the parts become: null, false, true. Or should we go opinionated, that true then false reads better in practice because it's easier to express & think in terms of passing than failing conditions?

Hmm. OTOH, those from a C-like background may find it more natural in the `F#` order, given that `false` is `0` and `true` is (usually) `1`, and then parameters are in order. I value consistency, but I'm not sure how many people use both F# and C#. I'm undecided.